pkg/util/cloudfoundry/types.go

```diff
@@ -202,3 +219,19 @@ func getVcapServicesMap(vcap, appGUID string) (map[string][]byte, error) {
 
 	return ret, nil
 }
+
+func getVcapApplicationMap(vcap string) (map[string]interface{}, error) {
```

Comment:
What about unmarshalling and storing exactly the fields that we need later? IMO it would be sufficient then to have the return value `map[string]string` with just the relevant information (on clusters with thousands of apps that might save some memory).